### ISSUE TITLE
xp bar location/dimensions

### ISSUE THREAD

**Issue by hushpuppi3** [SirHitech/MapleXPBar#4]
Would love it if there was a way to move the xp bar, for me specifically to the bottom of the screen (like classic MMOs)

Changing size would also be nice but less important

**Comment by H4waiianPunch** [comment]
> Would love it if there was a way to move the xp bar, for me specifically to the bottom of the screen (like classic MMOs)
> 
> Changing size would also be nice but less important

I second this for sure, the bar covers up my pop-out private chat just by a very small margin but it is incredibly annoying having the message covered.

**Comment by SirHitech** [comment]
Thanks for the suggestion and requests - terribly sorry it took me so long, I wasn't thrilled with my best efforts to implement it from a user experience perspective (ideally it's a draggable interface, but because it's a custom render, I wasn't sure how to). However this current implementation is better than none at all. Here it is in action:

![Image](https://github.com/user-attachments/assets/b21f5c57-5d52-4512-baac-a6d48a9eed95)

**Comment by H4waiianPunch** [comment]
Looking great! The offset X/Y should do just fine, no need for it to be draggable in this case, in my mind! 

As for the size, I think scaling it by a % is a smart way to do it, however, is there any chance we could also get options for X/Y scaling as well? I'd like to increase the height of mine to make it a bit easier to see, but I want it to stay on the chat popup so I don't want it to be any wider.

**Comment by SirHitech** [comment]
TL;DR - I will cook up a version without scaling size as a percentage that still allows for customizing the size of the bar. I expect to have something working this weekend that I'll post here for feedback.

Upon further testing, I think I will have to scrap sizing as a % in favor of resizing height/width independently. This is due to the fact that scaling the bar via a transform as I was doing also scales the displacement of the bar in an unfavorable manner. It's a bit hard to explain without visuals, so here's some videos demonstrating the issue.

When the bar is scaled, it's displaced quite a bit. This is somewhat fixed by adjusting the 'Offset Up/Down' values, however this doesn't function correctly when the chatbox is then hidden - in this case, the bar clips through the chatbox tabs. (Correct behavior of the bar 'sticking' to the hidden chatbox shown after, when values are reset to the default)

https://github.com/user-attachments/assets/b221cab3-0d93-456c-a09b-92c641c96935

Another issue is moving the chatbox around, the displacement scaling doesn't quite work and the bar floats off as the chatbox is moved around the screen.

https://github.com/user-attachments/assets/2d1d2e14-380d-44ba-bb92-0b40b04b294e

While it should theoretically be possible to un-apply the scaling strategically to account for the hiding and movement of the chatbox, it seems far more complicated than changing directions and going for individual height/width control as pixel values instead of the scaling percentage. I'll work on that iteration today/this weekend and post something here once it's ready.

**Comment by SirHitech** [comment]
Alright, here it is with thickness and length options instead of total % scaling, and I also threw in font size in case some people want to make it really big and need that larger. The bar now correctly anchors to the chatbox.

https://github.com/user-attachments/assets/084aa22d-80b4-42d4-b970-444bb4d157a3

And in the event people use the displacement stuff to move the bar off the chatbox, I added an option to decouple it and it won't move when the chatbox is closed or moved around on the screen. As an example, I've shrunk it down and placed it over the inventory instead.

![Image](https://github.com/user-attachments/assets/4ee36d05-7fd4-411f-a692-ee134524dcf8)

**Comment by SirHitech** [comment]
Feature is live, feel free to report any issues or feedback here. Enjoy!
